samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -298,14 +298,49 @@ def _build_lambda_function_code_property(tf_properties: dict) -> Any:
         return filename
 
     code = {}
-    tf_cfn_prop_names = [("s3_bucket", "S3Bucket"), ("s3_key", "S3Key")]
+    tf_cfn_prop_names = [("s3_bucket", "S3Bucket"), ("s3_key", "S3Key"), ("image_uri", "ImageUri")]
     for tf_prop_name, cfn_prop_name in tf_cfn_prop_names:
         tf_prop_value = tf_properties.get(tf_prop_name)
         if tf_prop_value is not None:
             code[cfn_prop_name] = tf_prop_value
     return code
 
 
+def _build_lambda_function_image_config_property(tf_properties: dict) -> Optional[dict]:
+    """
+    Builds the ImageConfig property of a CloudFormation AWS Lambda Function out of the
+    properties of the equivalent terraform resource
+
+    Parameters
+    ----------
+    tf_properties: dict
+        Properties of the terraform AWS Lambda function resource
+
+    Returns
+    -------
+    dict
+        The built ImageConfig property of a CloudFormation AWS Lambda Function resource
+    """
+    image_config = tf_properties.get("image_config")
+    if not image_config:
+        return None
+    image_config = image_config[0]
```

Comment:
do we need to assert that image_config type here, before attempting to get the first element.

Reply:
Primarily because we are relying on terraform not changing their contract on how they operate with their files, we should sanity check types of whatever fields we are accessing.

Reply:
as per the terraform plan output, it should be a list. Terraform translate Optional attributes into List, and either it is empty, or contain one item.

Reply:
as per our discussion, I will add this check to guard our code in case if terraform changed there contract, and to fail elegantly